### PR TITLE
211 finalize all repos to work with updated interfaces 1

### PR DESCRIPTION
**Pull request by lance-wilson**
# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
supports NRLMMD-GEOIPS/geoips#211
Also fixes NRLMMD-GEOIPS/geoips_plugin_example#2 (duplicates https://github.com/NRLMMD-GEOIPS/geoips_plugin_example/pull/3)

# Testing Instructions

Run tests/test_all.sh

# Summary

Tasks in progress:
* Switching interface_modules to plugins/modules
* Switching yaml_configs to plugins/yaml
* Switching to pyproject.toml
* Switching to CHANGELOG.rst
* Changing function names to "call"
* Removing VERSION file
* Formatting changes (black and flake8)

# Output

Returns 0 for all four tests.
(geoips_conda) ./tests/test_all.sh
Mon May 15 16:08:38 CDT 2023 Running pre, log: /home/lwilson/geoproc/test_data/logs/20230515/20230515.210838_geoips_plugin_example/test_all_geoips_plugin_example.log
Mon May 15 16:08:38 CDT 2023 Running check_code.sh
/home/lwilson/geoproc/geoips_packages/geoips/tests/utils/check_code.sh all ./tests/../
/home/lwilson/geoproc/test_data/logs/20230515/20230515.210838_geoips_plugin_example/test_all_geoips_plugin_example.log_check_code.sh__all_._tests_.._.log
Return: 0

Mon May 15 16:08:40 CDT 2023 Running test_interfaces
test_interfaces
/home/lwilson/geoproc/test_data/logs/20230515/20230515.210838_geoips_plugin_example/test_all_geoips_plugin_example.log_test_interfaces.log
Return: 0

Mon May 15 16:09:09 CDT 2023 Running pytest
pytest /home/lwilson/geoproc/geoips_packages/geoips/tests/test_pytest
/home/lwilson/geoproc/test_data/logs/20230515/20230515.210838_geoips_plugin_example/test_all_geoips_plugin_example.log_pytest___home_lwilson_geoproc_geoips_packages_geoips_tests_test_pytest.log
Return: 0

Mon May 15 16:09:28 CDT 2023 Running abi_aws.sh
/home/lwilson/geoproc/geoips_packages/geoips_plugin_example/tests/scripts/abi_aws.sh
/home/lwilson/geoproc/test_data/logs/20230515/20230515.210838_geoips_plugin_example/test_all_geoips_plugin_example.log_abi_aws.sh.log
Return: 0

Mon May 15 16:09:47 CDT 2023  Running post, final results in /home/lwilson/geoproc/test_data/logs/20230515/20230515.210838_geoips_plugin_example/test_all_geoips_plugin_example.log
Package: geoips_plugin_example
Total run time: 69 seconds
Number data types run: 4
Number data types failed: 0

### CODE REVIEW

File: geoips_plugin_example/__init__.py [hunk 1]
```diff
@@ -9,3 +9,4 @@
 # # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
 # # # for more details. If you did not receive the license, for more information see:
 # # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+"""Init file for geoips_plugin_example."""
```
Comment (mindyls) [inline geoips_plugin_example/__init__.py:12]: @lance-wilson This needs to have the "version" information like is contained in the main geoips repo __init__.py file.  Could you also please update the Issue to note that the version imports should be added to the top level __init__.py file?

Additionally, every directory that contains python files in any sub-directory must have an __init__.py file.  Maybe make a note of that in the main issue as well?

Comment (lance-wilson) [reply geoips_plugin_example/__init__.py:12]: Done.

Comment (mindyls) [reply geoips_plugin_example/__init__.py:12]: In [1]: import geoips_plugin_example

In [2]: geoips_plugin_example.__version__
Out[2]: '1.6.0.post24'

Looks good

File: geoips_plugin_example/__init__.py [hunk 2]
```diff
@@ -9,3 +9,8 @@
...
+
+from .version import __version__, __version_tuple__
+
+__all__ = ["__version__", "__version_tuple__"]
```
Comment (mindyls) [inline geoips_plugin_example/__init__.py:16]: @lance-wilson Note I added "__all__" so the code_checks passed (it was failing on unused __version__)

File: geoips_plugin_example/plugins/modules/colormaps/visir/__init__.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# # # Author:
+# # # Naval Research Laboratory, Marine Meteorology Division
+# # #
+# # # This program is free software: you can redistribute it and/or modify it under
+# # # the terms of the NRLMMD License included with this program. This program is
+# # # distributed WITHOUT ANY WARRANTY; without even the implied warranty of
+# # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+# # # for more details. If you did not receive the license, for more information see:
+# # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+"""Init file for geoips_plugin_example/colormaps/visir."""
```
Comment (mindyls) [inline geoips_plugin_example/plugins/modules/colormaps/visir/__init__.py:12]: @lance-wilson Added one more init file.

**Review by mindyls — APPROVED** [review]
With pip install (not pip install -e, so it checks that all init files are there), full success, and correct version import.

Package: geoips_plugin_example
Total run time: 103 seconds
Number data types run: 4
Number data types failed: 0


In [1]: import geoips_plugin_example

In [2]: geoips_plugin_example.__version__
Out[2]: '1.6.0.post25'

In [3]: geoips_plugin_example.__version_tuple__
Out[3]: (1, 6, 0)

**Pull request merged**